src/pyspiffe/workloadapi/default_workload_api_client.py

```diff
@@ -34,6 +37,51 @@
     WORKLOAD_API_HEADER_VALUE,
 )
 
+# GRPC Error Codes that the client will not retry on:
+#  - INVALID_ARGUMENT is not retried according to the SPIFFE spec because the request is invalid
+#  - CANCELLED is not retried because it occurs when the caller has canceled the operation.
+_NON_RETRYABLE_CODES = {grpc.StatusCode.CANCELLED, grpc.StatusCode.INVALID_ARGUMENT}
+
+
+class _RetryHandler:
+    def __init__(
+        self,
+        max_retries: int = 0,
```

Comment:
should max_retries be set to another valid number? Like 3... that makes more sense the 0 to me.. Or the intent is not to have retries by default?

Reply:
The intent is to have an undefined number of retries by default, so if `max_retries=0` there is no maximum.

Reply:
To make the intent clear I replaced this magic number "0" by a constant.